From ed2cbb82820dc05cf4969486f312bdbc79e212a2 Mon Sep 17 00:00:00 2001
From: frq09432 <nicolas.louboutin@st.com>
Date: Thu, 16 Jan 2020 08:42:19 +0100
Subject: [PATCH 1/4] drm_hwcomposer: Remove error log linked to layer import
 in case of tested composition

  The ImportBuffer function can return an error in test mode if a layer can't be composed (depending on usage).
  The log is then not required to avoid misunderstanding.

Change-Id: I0b805c1aa36378cbd2e34d556d09d292e2a2390a
---
 drmhwctwo.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drmhwctwo.cpp b/drmhwctwo.cpp
index ffc5fcd..95370d5 100644
--- a/drmhwctwo.cpp
+++ b/drmhwctwo.cpp
@@ -604,7 +604,8 @@ HWC2::Error DrmHwcTwo::HwcDisplay::CreateComposition(bool test) {
     l.second->PopulateDrmLayer(&layer);
     int ret = layer.ImportBuffer(importer_.get());
     if (ret) {
-      ALOGE("Failed to import layer, ret=%d", ret);
+      if (!test)
+        ALOGE("Failed to import layer, ret=%d", ret);
       return HWC2::Error::NoResources;
     }
     map.layers.emplace_back(std::move(layer));
-- 
2.7.4

